### PR TITLE
ref(sampling): Remove error rules code - [TET-103]

### PR DESCRIPTION
**Pull request by priscilawebdev**
As we are now only focusing on Dynamic Sampling for transactions and would like to rethink how Errors will be implemented on the page, we decided to remove its code until then.

What this PR does:

- Remove the feature flag `organizations:filters-and-sampling-error-rules`
- Remove error rules code
- Minor tests simplification in the file`/projectFiltersAndSampling/transactions.spec.tsx `

### CODE REVIEW

File: tests/js/spec/views/settings/projectFiltersAndSampling/transactions.spec.tsx [hunk 1]
```diff
@@ -142,69 +104,54 @@ describe('Filters and Sampling - Transaction rule', function () {
       expect(screen.getByLabelText('Search or add a release')).toBeInTheDocument();
 
       // Release field is not empty
-      const releaseFieldValues = screen.getByTestId('multivalue');
-      expect(releaseFieldValues).toBeInTheDocument();
-      expect(releaseFieldValues).toHaveTextContent('1.2.3');
```
Comment (matejminar) [inline tests/js/spec/views/settings/projectFiltersAndSampling/transactions.spec.tsx:147]: ❤️ love these simplifications

File: tests/js/spec/views/settings/projectFiltersAndSampling/transactions.spec.tsx [hunk 2]
```diff
@@ -552,16 +482,15 @@ describe('Filters and Sampling - Transaction rule', function () {
...
-            expect(checkedCheckbox).toBeChecked();
+            expect(screen.getByRole('checkbox')).toBeChecked();
 
             // Uncheck tracing checkbox
-            userEvent.click(checkedCheckbox);
+            userEvent.click(screen.getByRole('checkbox'));
 
             // Unched tracing checkbox
-            expect(checkedCheckbox).not.toBeChecked();
+            expect(screen.getByRole('checkbox')).not.toBeChecked();
```
Comment (matejminar) [inline tests/js/spec/views/settings/projectFiltersAndSampling/transactions.spec.tsx:493]: These might get a bit slower as we query the element multiple times. We can play with it in TET-81

**Review by matejminar — APPROVED** [review]
Looks good to me.

Should we move the backend changes to a separate PR?

File: static/app/views/settings/project/filtersAndSampling/filtersAndSampling.tsx [hunk 3]
```diff
@@ -240,25 +202,14 @@ class FiltersAndSampling extends AsyncView<Props, State> {
...
         <TextBlock>
           {t('Rules for traces should precede rules for individual transactions.')}
         </TextBlock>
         <RulesPanel
-          rules={transactionRules}
+          rules={rules}
           disabled={disabled}
-          onAddRule={this.handleAddRule('transactionRules')}
-          onEditRule={this.handleEditRule}
+          onAddRule={this.handleOpenRule()}
```
[the remark this answers is not in the document]
Comment (priscilawebdev) [reply static/app/views/settings/project/filtersAndSampling/filtersAndSampling.tsx:211]: yes, I wanted to to improve it but left it to another PR

File: tests/js/spec/views/settings/projectFiltersAndSampling/transactions.spec.tsx (hunk 2 above)
Comment (priscilawebdev) [reply tests/js/spec/views/settings/projectFiltersAndSampling/transactions.spec.tsx:493]: yes I thought we could tackle it on that other story

**Pull request merged**